github/copilot.go

```diff
@@ -0,0 +1,311 @@
+// Copyright 2023 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// CopilotService provides access to the Copilot-related functions
+// in the GitHub API.
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/
+type CopilotService service
+
+// CopilotOrganizationDetails represents the details of an organization's Copilot for Business subscription.
+type CopilotOrganizationDetails struct {
+	SeatBreakdown         *CopilotSeatBreakdown `json:"seat_breakdown"`
+	PublicCodeSuggestions string                `json:"public_code_suggestions"`
+	CopilotChat           string                `json:"copilot_chat"`
+	SeatManagementSetting string                `json:"seat_management_setting"`
+}
+
+// CopilotSeatBreakdown represents the breakdown of Copilot for Business seats for the organization.
+type CopilotSeatBreakdown struct {
+	Total               int `json:"total"`
+	AddedThisCycle      int `json:"added_this_cycle"`
+	PendingCancellation int `json:"pending_cancellation"`
+	PendingInvitation   int `json:"pending_invitation"`
+	ActiveThisCycle     int `json:"active_this_cycle"`
+	InactiveThisCycle   int `json:"inactive_this_cycle"`
+}
+
+// ListCopilotSeatsResponse represents the Copilot for Business seat assignments for an organization.
+type ListCopilotSeatsResponse struct {
+	TotalSeats int64                 `json:"total_seats"`
+	Seats      []*CopilotSeatDetails `json:"seats"`
+}
+
+// CopilotSeatDetails represents the details of a Copilot for Business seat.
+type CopilotSeatDetails struct {
+	// Assignee can either be a User, Team, or Organization.
+	Assignee                interface{} `json:"assignee"`
+	AssigningTeam           *Team       `json:"assigning_team,omitempty"`
+	PendingCancellationDate *string     `json:"pending_cancellation_date,omitempty"`
```

Comment:
type should be `*Timestamp` instead that it works in similar way than other time fields.